staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -80,6 +81,7 @@ func (s *ServerRunOptions) ApplyTo(c *server.Config) error {
 	c.JSONPatchMaxCopyBytes = s.JSONPatchMaxCopyBytes
 	c.MaxRequestBodyBytes = s.MaxRequestBodyBytes
 	c.PublicAddress = s.AdvertiseAddress
+	c.PublicPort = s.AdvertisePort
```

Comment:
I expected ServerRunOptions#DefaultAdvertiseAddress below to populate AdvertisePort from the secure serving port

Reply:
Done.